packages/ListBox/stories/moreExamples/directory/directory.stories.js

```diff
@@ -0,0 +1,163 @@
+    }
+  }, 500);
+
+  function handleChange(item, list) {
+    if (item === null) {
+      // means list was cleared
+      setSelected(() => null);
+      return;
+    }
+
+    setSelected(() => list[item].value);
+    setCustomValue(() => null);
+  }
+
+  function handleKeyDown(event) {
+    if (event.target.value !== "") {
+      setIsLoading(() => true);
+    } else {
+      setIsLoading(() => false);
+    }
+
+    if (event.key === "Enter" && event.target.value !== "") {
+      event.stopPropagation();
+      if (!isLoading && showNoResults) {
+        const label = event.target.value;
+        setSelected(label);
+        setCustomValue(() => ({ label }));
+        setShowNoResults(() => false);
+        setIsLoading(() => false);
+        initialValues(allUsers, allContacts);
+        refFilterInput.current.clear();
+        refListBox.current.close();
+        window.requestAnimationFrame(() => {
+          refInputName.current.focus();
+        });
+      }
+    }
+  }
+
+  /* eslint-disable jsx-a11y/label-has-associated-control */
+  return (
+    <>
+      <ListBox ref={refListBox} isDisabled={isListBoxDisabled} onChange={handleChange} placeholder="Select user...">
+        <ListBox.Filter ref={refFilterInput} onKeyDown={handleKeyDown} filter={filter} />
+        <>
+          {isLoading && !showNoResults ? (
+            <ListBox.RawItem isDisabled>
+              <Spinner size="small" />
+            </ListBox.RawItem>
+          ) : null}
+          {showNoResults ? (
+            <ListBox.RawItem isDisable>
+              <span>No Results. Hit Enter to use Jack Johnson as *owner* name</span>
+            </ListBox.RawItem>
+          ) : null}
+          {!showNoResults && customValue ? (
+            <ListBox.Option defaultIsSelected={customValue.label} value={customValue.label}>
+              {customValue.label}
+            </ListBox.Option>
+          ) : null}
+          {!showNoResults && contacts.length ? <ListBox.Divider key="divider_2">Contacts</ListBox.Divider> : null}
+          {!showNoResults &&
+            contacts.map(contact => (
+              <ListBox.Option defaultIsSelected={selected === contact.label} value={contact.label} key={contact.label}>
+                {contact.label}
+              </ListBox.Option>
+            ))}
+          {!showNoResults && users.length ? <ListBox.Divider key="divider_1">Users</ListBox.Divider> : null}
+          {!showNoResults &&
+            users.map(user => (
+              <ListBox.Option defaultIsSelected={selected === user.label} value={user.label} key={user.label}>
+                {user.label}
+              </ListBox.Option>
+            ))}
+        </>
+      </ListBox>
+
+      {customValue ? (
+        <>
+          <hr />
+          <label>
+            Email: <input ref={refInputName} type="text" defaultValue="" />
+          </label>
+        </>
+      ) : null}
+    </>
+  );
+}
+
+storiesOf("ListBox / Lazy", module).add("Directory", () => (
```

Comment:
This should really be under a folder called "backyard" But no worries can clean up later